lib/dry/view/decorator.rb

```diff
@@ -27,6 +28,15 @@ def call(name, value, renderer:, context:, **options)
       def part_class(name, value, **options)
         options.fetch(:as) { Part }
       end
+
+      private
+
+      # @api private
+      def singularize_options(**options)
+        options = options.dup
```

Comment:
when you capture values with `**` ruby creates a new hash instance so calling `.dup` is not needed

```ruby
2.3.4 :001 > def foo(**options) options.object_id; end
 => :foo
2.3.4 :002 > {}.tap { |h| puts foo(h) == h.object_id }
false
```